### PR TITLE
fix(ImagingRequest): imaging table requestedBy show current usr

### LINKED ISSUE #2540: Imaging Request - requestedBy should store the currently logged in user id
## 🐛 Bug Report

Imaging request is currently hardcoded to set `requestedBy` to **test** instead of the currently logged in user's id. This needs to be corrected to user the current user id from the redux store.

## To Reproduce

- Access staging https://staging.hospitalrun.io/
- Create a patient and add a visit to this patient
- Create an imaging request for the above patient/visit
- View the list of imaging requests and notice that the requested by field shows 'test'

## Expected behavior

- The `requestedBy` should be the currently logged in `user.id` at the time the imaging request is created
- The list of imaging requests should display the `user.fullName` based on the imaging request's `requestedBy`

## Notes
- See `src/shared/components/navbar/Navbar.tsx` on how to access the user from the store

## Your Environment

- *node version*: 6,8,10
- *fastify version*: >=1.0.0
- *os*: Mac, Windows, Linux
- *any other relevant information*

### PR DESCRIPTION
**Pull request by riiniii**
requestedBy's current user show's 'test', not actual log in user, pass in logged in user from redux
store to useRequestImaging hook

Fixes #2540.

**Changes proposed in this pull request:**

- pass in user information to useRequestImaging hook from NewImagingRequest page by using user information from redux store
- export interfaces in user-slice to type in tests

**Newly added dependencies with [Bundlephobia](https://bundlephobia.com/) links:**

- no new dependencies

### CODE REVIEW

File: src/imagings/hooks/useRequestImaging.tsx [hunk 1]
```diff
@@ -14,22 +14,24 @@ export interface ImagingRequest {
...
+    }
 
-  await ImagingRepository.save({
-    ...request,
-    requestedBy: 'test',
-    requestedOn: new Date(Date.now()).toISOString(),
-  } as Imaging)
+    await ImagingRepository.save({
+      ...request,
+      requestedBy: user?.fullName || '',
```
Comment (blestab) [inline src/imagings/hooks/useRequestImaging.tsx:27]: Hi @riiniii 
Note the requirements  
- _"The requestedBy should be the currently logged in user.id at the time the imaging request is created"_
- _"The list of imaging requests should display the user.fullName based on the imaging request's requestedBy"_

So in other words we must store the user id (not full name as is currently in this code) then when displaying the imaging request, we display the full username. 

image

**Review by blestab — CHANGES_REQUESTED** [review]

File: src/imagings/hooks/useRequestImaging.tsx (hunk 1 above)
Comment (riiniii) [reply src/imagings/hooks/useRequestImaging.tsx]: Hi @blestab, thank you for the clarification. I should have asked for clarifications ^^'

I ended up passing requestedByFullName alongside requestedBy when saving a new imaging request, and showing the fullName in the table.

**Review by blestab — APPROVED** [review]
LGTM

File: src/imagings/hooks/useRequestImaging.tsx (hunk 1 above)
Comment (blestab) [reply src/imagings/hooks/useRequestImaging.tsx]: Okay great! Thanks

**Pull request merged**